packages/xrpl/src/Wallet/index.ts

```diff
@@ -440,6 +447,39 @@ class Wallet {
 
       return memo
     })
+
+    Object.keys(txCopy).forEach(function (key) {
+      const amount: IssuedCurrencyAmount = txCopy[key]
+      const standard_currency_code_len = 3
+      if (amount && isIssuedCurrency(amount)) {
+        const decodedCurrency = (decoded[key] as unknown as IssuedCurrencyAmount).currency
+
+        // Check for potential scams and alert the user
+        if (
+          amount.currency.length === standard_currency_code_len &&
+          amount.currency !== amount.currency.toUpperCase()
+        ) {
+          let errorMessage = `The standard currency code provided (${amount.currency}) has lowercase letters and may be impersonating a corresponding ISO currency.`
+          if (amount.currency.toUpperCase() === 'XRP') {
+            errorMessage = `The standard currency code provided (${amount.currency}) to sign a transaction involving an issued currency with a similar name to 'XRP'.`
+                            + `This is likely a scam. XRP is not an issued currency.`
+          }
+          const hex = isoToHex(amount.currency)
+          const workAround = `If this is intentional, please use the 40 byte hex encoding of this currency code instead of the standard format (${amount.currency} -> ${hex}).`
+          throw new XrplError(errorMessage + " " + workAround)
+        }
+
+        // Then standardize the format of currency codes to the 40 byte hex string for comparison
+        if(amount.currency.length !== decodedCurrency.length) {
+          if(decodedCurrency.length == standard_currency_code_len) {
+            (decoded[key] as unknown as IssuedCurrencyAmount).currency = isoToHex(decodedCurrency)
+          } else {
+            txCopy[key].currency = isoToHex(txCopy[key].currency)
+          }
+        }
+      }
+    })
```

Comment:
Hmm... I don't really like how this implementation adds a performance hit to all signing in exchange for protecting against a rare edge case. 

Alternatives:
1. Enforce this at the `decode` level in `ripple-binary-codec` - That way there's no additional lookup required. (Before this, ripple-binary-codec didn't allow lowercase anyway, this could just be a better error message)
2. Add it as a check when creating an IssuedCurrencyAmount potentially?

Thoughts? @khancode @intelliot

Reply:
Regardless of the currency code, users have to trust the accounts that they open trustlines with. (Hence the name "trust lines".) "usd" could legitimately be used for US Dollars, if issuers and holders agreed to it and creditors understood the counterparty risk.

In https://github.com/XRPLF/xrpl.js/issues/1528 the problem is that the currency code codec is not symmetric: encoding "usd" yields one result, but decoding that result doesn't yield "usd". 3-letter uppercase currency codes made sense 5 years ago, but we are now in a world where even currencies like EUR are being issued with non-standard codes (`EURS`). Because of this, I think our libraries should change with the times.

We have two options:

1. Make the codec symmetric so that "usd" in becomes "usd" out. This is a breaking change unless we leave the existing behavior and add a new method/option for this.
2. Require that lowercase or symbol-containing "Standard" currency codes are provided in hex (similar to what we do for [NFTokenMint.URI](https://github.com/XRPLF/xrpl.js/pull/2008#discussion_r884002507)) - that is what we are currently doing, and we're improving the error messages. If we choose this option, then we should add a method to make it easy to convert to/from hex, and make sure this method is documented and used in tutorials.

Reply:
After talking through this a bit, I'm leaning towards making the codec symmetric and allowing lowercase standard currency codes in xrpl.js

The reason being that folks can already issue capitalized standard currencies to pretend to be legitimate (Ex. "USD"). So like @intelliot pointed out above - users already have to determine if the issued currency is trustworthy by checking the issuer since the currency code offers no protections (They can name it whatever they want besides XRP). 

So, to resolve #1528, I think we should just update the binary-codec to decode lowercase and allowed symbol standard currency codes. This fix should NOT include additional checks in xrpl.js like I implemented here to disallow lowercase standard currency codes.

@mDuo13 Does that resolution seem reasonable?

Reply:
I think it's fine to allow lowercase standard currency codes. We shouldn't be overly aggressive in what we limit, because there _are_ legitimate usecases that someone might want to do that.

Reply:
Just curious - what is a legitimate use case where someone would want to do that?

ISO 4217 codes are all capital letters, AFAIK.

Reply:
There are all sorts of tokens on the ledger right now with various names and capitalizations. I figured there's gotta be at least one that has 3 lower-case letters.

Reply:
One I found that is 3 characters and has at least one lowercase is xSD.  Not to be confused with XSD which is also on the ledger.

Reply:
Agree that they exist and are definitely there. Not sure that they are a legitimate use case. Would any of them be fundamentally worse if they used all capital letters instead?